xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -439,10 +439,21 @@ void CSelectionStreams::Update(const std::shared_ptr<CDVDInputStream>& input,
     std::string filename = nav->GetFileName();
     int source = Source(STREAM_SOURCE_NAV, filename);
 
+    std::vector<CDemuxStream*> demuxStreams;
+    if (demuxer)
+      demuxStreams = demuxer->GetStreams();
+
     int count;
     count = nav->GetAudioStreamCount();
     for(int i=0;i<count;i++)
     {
+      const auto stream =
+          std::find_if(demuxStreams.begin(), demuxStreams.end(),
+                       [i](const auto& stream)
+                       { return stream->type == STREAM_AUDIO && stream->dvdNavId == i; });
+      CDemuxStreamAudio* aStream =
+          (stream != demuxStreams.end()) ? static_cast<CDemuxStreamAudio*>(*stream) : nullptr;
+
```

Comment:
this can be full removed not really needed,
imo compare "dvdNavId" id to an index count could be not so good idea

Reply:
The dvdNavId business is a little hard to follow but necessary to connect the ffmpeg audio streams list with the dvd nav audio stream info. I experienced mismatches when relying on stream enumeration order alone because the ffmpeg streams list gets shuffled occasionally and doesn't match the physical stream order anymore., while dvdnav info uses the physical order. VP has that dvdnavid to maintain the link between the two lists.

So, ordinal in dvdnav audio stream enumeration (i in the loop) = dvdnavid of the matching ffmpeg stream.
You can check other code in VP using dvdnavid and m_dvd. I can add a comment to explain some of this.

The IFO files contain a sample rate (48k or 96k), but not a bitrate from what I could see. They don't have a value defined for dts-es and sometimes an incorrect channel count, which is why I went looking for the info per ffmpeg. But it's a separate commit so it could be omitted if "the team" is not comfortable.
see http://www.mpucoder.com/DVD/ifo.html, "audio attributes" libdvdnav/read headers and code, dvd_types.h

Reply:
i understood interesting link
so anyway i find a good thing add a little comment something like example
`// when possible prefer ffmpeg info because more reliable than IFO nav metadata`